### PR TITLE
Hardening on background tasks and issues search - 2016-09-12

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/ce/ws/ActivityAction.java [hunk 1]
```diff
@@ -280,6 +280,21 @@ private static Long parseDateTimeAsLong(@Nullable String dateAsString) {
...
+      return null;
+    }
+
+    Date date = parseDateTimeQuietly(dateAsString);
+    if (date == null) {
+      date = parseDateQuietly(dateAsString);
+      checkRequest(date != null, "Date '%s' cannot be parsed as either a date or date+time", dateAsString);
+    }
+
+    return date.getTime();
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/ce/ws/ActivityAction.java]: MAJOR NullPointerException might be thrown as 'date' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by julienlancelot** [comment]
The commit message "	
SONAR-8028 WS issues/search from and to inclusive fro data and date+time" should be fixed (fro -> for)

File: server/sonar-server/src/main/java/org/sonar/server/ce/ws/ActivityAction.java [hunk 2]
```diff
@@ -280,6 +280,21 @@ private static Long parseDateTimeAsLong(@Nullable String dateAsString) {
...
+  private static Long parseFromDate(@Nullable String dateAsString) {
+    if (dateAsString == null) {
+      return null;
+    }
+
+    Date date = parseDateTimeQuietly(dateAsString);
+    if (date == null) {
+      date = parseDateQuietly(dateAsString);
+      checkRequest(date != null, "Date '%s' cannot be parsed as either a date or date+time", dateAsString);
+    }
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/ce/ws/ActivityAction.java]: I don't like having so much duplication... I don't see how we could do, but I don't like it...

Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/ce/ws/ActivityAction.java]: I agree, I'll refactor later on (when handling QP Changelog). I'll add 2 methods in DateUtils: `parseEndingDate(String)` and `parseStartingDate(String)`

**Comment by julienlancelot** [comment]
LTGM

**Pull request merged**